pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/pubsub/MercuryOrderStatusListener.java

```diff
@@ -11,26 +11,33 @@
 import com.google.gson.Gson;
 import com.google.pubsub.v1.ProjectSubscriptionName;
 import com.google.pubsub.v1.PubsubMessage;
+import lombok.extern.slf4j.Slf4j;
 import org.broadinstitute.dsm.db.dao.mercury.MercuryOrderDao;
+import org.broadinstitute.dsm.exception.DSMPubSubException;
 import org.broadinstitute.dsm.model.mercury.BaseMercuryStatusMessage;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
+@Slf4j
 public class MercuryOrderStatusListener {
-    private static final Logger logger = LoggerFactory.getLogger(MercuryOrderStatusListener.class);
 
+    private MercuryOrderStatusListener(){
+        throw new IllegalStateException("Utility class");
+    }
     public static void subscribeToOrderStatus(String projectId, String subscriptionId) {
         // Instantiate an asynchronous message receiver.
         MessageReceiver receiver = (PubsubMessage message, AckReplyConsumer consumer) -> {
+            String messageId = message.getMessageId();
             // Handle incoming message, then ack the received message.
-            logger.info("Got STATUS message with Id: " + message.getMessageId());
-
+            log.info(String.format("Got STATUS message with Id: %s", messageId));
             try {
+                consumer.ack();
                 processOrderStatus(message);
-                logger.info("Processing the message finished");
+                log.info("Processing the message finished");
+            } catch (DSMPubSubException error) {
+                log.error("Error happened parsing Mercury Status Message, DSM will ack the message", error);
                 consumer.ack();
+                error.printStackTrace();
             } catch (Exception ex) {
-                logger.info("about to nack the message", ex);
+                log.error("Unexpected error: about to nack the message", ex);
                 consumer.nack();
```

Comment:
The error message should have some context about which message (not necessarily the content, but the message type, etc.).
This nack is now unnecessary.